### PR TITLE
Eng 7612 ee crash message to log

### PR DESCRIPTION
**Pull request by jxchinaren**

### CODE REVIEW

File: src/ee/expressions/functionexpression.cpp [hunk 1]
```diff
@@ -35,6 +35,15 @@ template<> inline NValue NValue::callUnary<FUNC_VOLT_SQL_ERROR>() const {
         const int32_t valueLength = getObjectLength_withoutNull();
         const char *valueChars = reinterpret_cast<char*>(getObjectValue_withoutNull());
         std::string valueStr(valueChars, valueLength);
+
+#if defined(DEBUG) || defined(_DEBUG) || defined(_DEBUG_)
+        if (valueStr == "crash_me") {
```
Comment (wolffcm) [inline src/ee/expressions/functionexpression.cpp]: Can you do a comparison of an std::string to a const char* literal like this?  I have the feeling if this compiles that it always returns false?  

(Actually I just verified that the C++ library includes operator overloads to make this work right, so I just learned something.)

File: src/ee/expressions/functionexpression.cpp [hunk 2]
```diff
@@ -35,6 +35,15 @@ template<> inline NValue NValue::callUnary<FUNC_VOLT_SQL_ERROR>() const {
         const int32_t valueLength = getObjectLength_withoutNull();
         const char *valueChars = reinterpret_cast<char*>(getObjectValue_withoutNull());
         std::string valueStr(valueChars, valueLength);
+
+#if defined(DEBUG) || defined(_DEBUG) || defined(_DEBUG_)
```
Comment (wolffcm) [inline src/ee/expressions/functionexpression.cpp]: There is also the more standard `#ifndef NDEBUG` for this kind of thing.  (This is what the C library's `assert` uses.)  In a couple of places we use that too.  I'm not sure which one is better.

File: src/ee/expressions/functionexpression.cpp [hunk 3]
```diff
@@ -35,6 +35,15 @@ template<> inline NValue NValue::callUnary<FUNC_VOLT_SQL_ERROR>() const {
...
+        if (valueStr == "crash_me") {
+            // generate a segmentation fault on purpose
+            int *ptr = NULL;
+            *ptr = 1; // EXC_BAD_ACCESS
```
Comment (wolffcm) [inline src/ee/expressions/functionexpression.cpp]: I think writing to a null pointer is undefined behavior.  This likely does what you want, but may not be portable, or may not do what you want for -O3, or when the compiler is upgraded.  The most portable way is probably to call the C function `raise(SIGSEGV)`.

I'm probably being pedantic; feel free to ignore.

**Comment by wolffcm** [comment]
Looks good to me.

**Comment by paulmartel** [comment]
Rebasing branch with changes re:wolffcm's comments.

**Pull request closed without merging**